optuna/integration/allennlp.py

```diff
@@ -128,3 +130,45 @@ def run(self) -> float:
 
         metrics = json.load(open(os.path.join(self._serialization_dir, "metrics.json")))
         return metrics[self._metrics]
+
+
+@experimental("2.0.0")
+class AllenNLPPruningCallback(EpochCallback):
+    """AllenNLP callback to prune unpromising trials.
+
+    See `the example <https://github.com/optuna/optuna/blob/master/
+    examples/allennlp/allennlp_simple.py>`__
+    if you want to add a proning callback which observes a metric.
+
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of the
+            objective function.
+        monitor:
+            An evaluation metric for pruning, e.g. ``validation_loss`` or
+            ``validation_accuracy``.
+    """
+
+    def __init__(self, trial: optuna.trial.Trial, monitor: str):
+        _imports.check()
+
+        if allennlp.__version__ < "1.0.0":
+            raise Exception("AllenNLPPruningCallback requires `allennlp`>=1.0.0.")
+
+        self._trial = trial
+        self._monitor = monitor
+
+    def __call__(
+        self,
+        trainer: "allennlp.training.GradientDescentTrainer",
+        metrics: Dict[str, Any],
+        epoch: int,
+        is_master: bool,
+    ) -> None:
+        value = metrics.get(self._monitor)
+        if not value:
```

Comment:
Not sure, `value` can be `None`, right?
```suggestion
        if value is None:
```

Reply:
You're right. Thank you!